x-pack/legacy/plugins/ml/public/data_frame_analytics/common/analytics.ts

```diff
@@ -12,16 +12,45 @@ import { Subscription } from 'rxjs';
 // @ts-ignore
 import { isJobIdValid } from '../../../common/util/job_utils';
 
-// TODO
-export const moveToAnalyticsWizard = () => {};
-
 export const isAnalyticsIdValid = isJobIdValid;
 
 export type IndexName = string;
 export type IndexPattern = string;
 export type DataFrameAnalyticsId = string;
 
-export interface DataFrameAnalyticsOutlierConfig {
+interface OutlierAnalysis {
+  outlier_detection: {};
+}
+
+interface RegressionAnalysis {
+  regression: {
+    dependent_variable: string;
+    training_percent?: number;
+  };
+}
+
+interface GenericAnalysis {
+  [key: string]: Record<string, any>;
+}
+
+type AnalysisConfig = OutlierAnalysis | RegressionAnalysis | GenericAnalysis;
+
+export const getAnalysisType = (analysis: AnalysisConfig) => {
+  const keys = Object.keys(analysis);
+
+  if (keys.length === 1) {
+    return keys[0];
+  }
+
+  return 'unknown';
+};
+
+export const isOutlierAnalysis = (arg: any): arg is OutlierAnalysis => {
+  const keys = Object.keys(arg);
+  return keys.length === 1 && keys[0] === 'outlier_detection';
```

Comment:
Could values for the analysis type `outlier_detection`, `unknown` etc line in a constants file?